arviz/plots/lmplot.py

```diff
@@ -0,0 +1,340 @@
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
+
+
+def plot_lm(
+    y,
+    idata=None,
+    x=None,
+    y_model=None,
+    y_hat=None,
+    num_samples=50,
+    kind_pp="samples",
+    kind_model="lines",
+    xjitter=False,
+    plot_dim=None,
+    backend=None,
+    y_kwargs=None,
+    y_hat_plot_kwargs=None,
+    y_hat_fill_kwargs=None,
+    y_model_plot_kwargs=None,
+    y_model_fill_kwargs=None,
+    backend_kwargs=None,
+    show=None,
+    figsize=None,
+    textsize=None,
+    axes=None,
+    legend=True,
+    grid=True,
+):
+    """Posterior predictive and mean plots for regression-like data.
+
+    Parameters
+    ----------
+    y : str or DataArray or ndarray
+        If str, variable name from observed_data
+    idata : InferenceData, Optional
+        Optional only if y is not str
+    x : str, tuple of strings, DataArray or array-like, optional
+        If str or tuple, variable name from constant_data
+        If ndarray, could be 1D, or 2D for multiple plots
+        If none, coords name of y (y should be DataArray).
+    y_model : str or Sequence, Optional
+        If str, variable name from posterior.
+        Its dimensions should be same as y plus added chains and draws.
+    y_hat : str, Optional
+        If str, variable name from posterior_predictive.
+        Its dimensions should be same as y plus added chains and draws.
+    num_samples : int, Optional, Default 50
+        Significant if `kind_pp` is "samples" or `kind_model` is "lines".
+        Number of samples to be drawn from posterior predictive or
+    kind_pp : {"samples", "hdi"}, Default "samples"
+        Options to visualize uncertainty in data.
+    kind_model : {"lines", "hdi"}, Default "lines"
+        Options to visualize uncertainty in mean of the data.
+    plot_dim : str, Optional
+        Necessary if y is multidimensional.
+    backend : str, Optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    y_kwargs : dict, optional
+        Passed to :meth:`mpl:matplotlib.axes.Axes.plot` in matplotlib
+        and :meth:`bokeh:bokeh.plotting.Figure.circle` in bokeh
+    y_hat_plot_kwargs : dict, optional
+        Passed to :meth:`mpl:matplotlib.axes.Axes.plot` in matplotlib
+        and :meth:`bokeh:bokeh.plotting.Figure.circle` in bokeh
+    y_hat_fill_kwargs : dict, optional
+        Passed to az.plot_hdi()
+    y_model_plot_kwargs : dict, optional
+        Passed to :meth:`mpl:matplotlib.axes.Axes.plot` in matplotlib
+        and :meth:`bokeh:bokeh.plotting.Figure.line` in bokeh
+    y_model_fill_kwargs : dict, optional
+        Significant if `kind_model` is "hdi". Passed to :func:`~arviz.plot_hdi`
+    backend_kwargs : dict, optional
+        These are kwargs specific to the backend being used. For additional documentation
+        check the plotting method of the backend.
```

Comment:
the links to the methods where these are passed are still missing, you can take them from #1728 to avoid having to track them down in the source code.